### PR TITLE
Add Rust implementation

### PR DESCRIPTION
**Pull request by izn**
Hey ya. :cat2: 

Here we add code for Music Theory dojo written in Rust. :notes: 

<p align="center">
  
</p>

### CODE REVIEW

File: music-theory/rust/Cargo.toml [hunk 1]
```diff
@@ -0,0 +1,11 @@
+[package]
+name = "music"
+version = "0.1.0"
+authors = ["Thaynã B. Moretti <moretty@gmail.com>"]
+edition = "2018"
```
Comment (gustavodiel) [inline music-theory/rust/Cargo.toml]: Now that's old

File: music-theory/rust/src/main.rs [hunk 2]
```diff
@@ -0,0 +1,259 @@
...
+            major_scale(Note::B),
+            ["B", "C#", "D#", "E", "F#", "G#", "A#"]
+        );
+
+        assert_eq!(
+            minor_scale(Note::B),
+            ["B", "C#", "D", "E", "F#", "G", "A"]
+        );
+    }
+}
```
Comment (gustavodiel) [inline music-theory/rust/src/main.rs]: Could you pretty please move those specs to a different file? thanks :3

File: music-theory/rust/Cargo.toml (hunk 1 above)
Comment (izn) [reply music-theory/rust/Cargo.toml]: This is the last Rust [edition](https://doc.rust-lang.org/edition-guide/editions/index.html) :confounded:

File: music-theory/rust/src/main.rs (hunk 2 above)
Comment (izn) [reply music-theory/rust/src/main.rs]: I couldn't agree more, but...

> **Unit Tests**
The purpose of unit tests is to test each unit of code in isolation from the rest of the code to quickly pinpoint where code is and isn’t working as expected. You’ll put unit tests in the src directory in each file with the code that they’re testing. The convention is to create a module named tests in each file to contain the test functions and to annotate the module with cfg(test).
>
> Reference: https://doc.rust-lang.org/book/ch11-03-test-organization.html

:confused:

**Pull request merged**